### PR TITLE
Note about upgrade, Closes #229

### LINKED ISSUE #229: Add note v2.0 needs to be installed from scratch (no update possible)

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #229

### CODE REVIEW

File: user-guide/lisk-core.md [hunk 1]
```diff
@@ -231,7 +231,10 @@ _See code: [dist/commands/core/uninstall.ts](https://github.com/LiskHQ/lisk-sdk/
 
 ## lisk core:upgrade NAME
 
-Upgrade an instance of Lisk Core to a specified or latest version.
+Upgrade an instance of Lisk Core(Binary) to a specified or latest version.
```
Comment (ghost) [inline user-guide/lisk-core.md]:
```suggestion
Upgrade an instance of Lisk Core (Binary) to a specified or latest version.
```

File: user-guide/lisk-core.md [hunk 2]
```diff
@@ -231,7 +231,10 @@ _See code: [dist/commands/core/uninstall.ts](https://github.com/LiskHQ/lisk-sdk/
...
+
+> __Note:__ The `core:upgrade` command is only supported for Lisk Core version `v2.0.0` and higher.<br>
```
Comment (ManuGowda) [inline user-guide/lisk-core.md]: The `core:upgrade` command is only supported for Lisk Core version `>v2.0.0`

Comment (ghost) [reply user-guide/lisk-core.md]: Don't we usually use version numbers (`2.0.0`) instead of tags (`v2.0.0`)?

File: user-guide/lisk-core.md [hunk 3]
```diff
@@ -231,7 +231,10 @@ _See code: [dist/commands/core/uninstall.ts](https://github.com/LiskHQ/lisk-sdk/
...
+> To upgrade Lisk Core(Binary) `v1.6.0` or lower versions, please use the [bash script](../../lisk-core/setup/binary.md#option-b-the-bash-script), or [remove](#lisk-coreuninstall-name) the old version and make a fresh [install](#lisk-coreinstall-name).
```
Comment (ghost) [inline user-guide/lisk-core.md]:
```suggestion
> To upgrade Lisk Core (Binary) `v1.6.0` or lower versions, please use the [bash script](../../lisk-core/setup/binary.md#option-b-the-bash-script), or [remove](#lisk-coreuninstall-name) the old version and make a fresh [install](#lisk-coreinstall-name).
```

**Review by ghost — CHANGES_REQUESTED** [review]
One important point that's missing is that lisk-commander installations and "legacy" installations (done using `installLisk.sh`) are not compatible with each other: i.e if you install with `installLisk.sh install [...]` you must upgrade with `installLisk.sh upgrade [...]`.

**Review by ghost — APPROVED** [review]

**Review by ManuGowda — APPROVED** [review]

**Pull request merged**